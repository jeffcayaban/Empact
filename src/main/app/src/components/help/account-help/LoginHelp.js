import React from "react";
import {Col, Grid, Panel, Row} from "react-bootstrap";
import {Link} from "react-router-dom";

import LoginPage from '../../../other/images/login-page.png';
import LoginLink from '../../../other/images/login-link.png';
import SuccessfulLoginDialog from "../../../other/images/successful-login.png";
import UnsuccessfulLoginDialog from "../../../other/images/unsuccessful-dialog.png";
import BackButton from "../../common/BackButton";

/**
 * Displays help information for logging into an account.
 */

class LoginHelp extends React.PureComponent {

    render() {
        return (
            <div className={'secondaryPageContainerStyle'}>
                <div className={"centerStyle"}>
                    <h2>Logging In</h2>
                    <p>Understand how to log in into you account.</p>
                </div>
                <div className={"titleContentPadding"}>
                    <Grid className={"divNoPadding"}>
                        <Row>
                            <Col md={12} lg={12}>
                                <div>
                                    <h4 style={{ margin: 0 }}>How to login</h4>
                                    <hr className={"helpHr"} />

                                    <p>
                                        1. Go to the <Link to={"/login"}>login page</Link>. This link is found on the top
                                        navigation bar.
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingTop: '20px', paddingBottom: '20px' }}>
                                        <img style={{ maxWidth: '20%', minWidth: '200px', align: 'center' }}
                                             src={LoginLink}
                                             alt="Login Link"
                                        />
                                    </div>

                                    <p>
                                        2. On this page, enter the username and password of the account you want to
                                        login to. Then click on the "Login" button to login into the account.
                                    </p>

                                    <div className={"centerStyle"} style={{ paddingTop: '20px', paddingBottom: '20px' }}>
                                        <img style={{ maxWidth: '40%', minWidth: '350px', align: 'center' }}
                                             src={LoginPage}
                                             alt="Login Page"
                                        />
                                    </div>

                                    <p>
                                        3. If the login is successful then you will be shown a confirmation notification,
                                        otherwise an error screen will be displayed.
                                    </p>


                                    <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                        <Grid className={"divNoPadding"}>
                                            <Row>
                                                <Col md={6} lg={6}>
                                                    <Panel>
                                                        <Panel.Heading>
                                                            <Panel.Title>
                                                                If successful
                                                            </Panel.Title>
                                                        </Panel.Heading>
                                                        <Panel.Body className={"centerStyle"}>
                                                            <img style={{ width: '300px', align: 'center' }}
                                                                 src={SuccessfulLoginDialog}
                                                                 alt="Login successful dialog"
                                                            />
                                                        </Panel.Body>
                                                    </Panel>
                                                </Col>
                                                <Col md={6} lg={6}>
                                                    <Panel>
                                                        <Panel.Heading>
                                                            <Panel.Title>
                                                                If unsuccessful
                                                            </Panel.Title>
                                                        </Panel.Heading>
                                                        <Panel.Body className={"centerStyle"}>
                                                            <img style={{ width: '320px', align: 'center' }}
                                                                 src={UnsuccessfulLoginDialog}
                                                                 alt="Login unsuccessful dialog"
                                                            />
                                                        </Panel.Body>
                                                    </Panel>
                                                </Col>
                                            </Row>
                                        </Grid>
                                    </div>
                                </div>
                                <BackButton link={"/help/account"}/>
                            </Col>
                        </Row>
                    </Grid>
                </div>
            </div>
        );
    }
}

export default LoginHelp;