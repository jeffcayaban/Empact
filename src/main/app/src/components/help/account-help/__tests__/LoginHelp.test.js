import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import LoginHelp from "../LoginHelp";

describe('LoginHelp', () => {

    it('should render successfully', () => {
        const component = shallow(<LoginHelp />);
        expect(toJson(component)).toMatchSnapshot();
    });
});
