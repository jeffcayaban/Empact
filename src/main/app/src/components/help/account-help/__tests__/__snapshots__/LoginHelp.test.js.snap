// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginHelp should render successfully 1`] = `
<div
  className="secondaryPageContainerStyle"
>
  <div
    className="centerStyle"
  >
    <h2>
      Logging In
    </h2>
    <p>
      Understand how to log in into you account.
    </p>
  </div>
  <div
    className="titleContentPadding"
  >
    <Grid
      bsClass="container"
      className="divNoPadding"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={12}
          md={12}
        >
          <div>
            <h4
              style={
                Object {
                  "margin": 0,
                }
              }
            >
              How to login
            </h4>
            <hr
              className="helpHr"
            />
            <p>
              1. Go to the 
              <Link
                replace={false}
                to="/login"
              >
                login page
              </Link>
              . This link is found on the top navigation bar.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <img
                alt="Login Link"
                src="login-link.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "20%",
                    "minWidth": "200px",
                  }
                }
              />
            </div>
            <p>
              2. On this page, enter the username and password of the account you want to login to. Then click on the "Login" button to login into the account.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingBottom": "20px",
                  "paddingTop": "20px",
                }
              }
            >
              <img
                alt="Login Page"
                src="login-page.png"
                style={
                  Object {
                    "align": "center",
                    "maxWidth": "40%",
                    "minWidth": "350px",
                  }
                }
              />
            </div>
            <p>
              3. If the login is successful then you will be shown a confirmation notification, otherwise an error screen will be displayed.
            </p>
            <div
              className="centerStyle"
              style={
                Object {
                  "paddingTop": "15px",
                }
              }
            >
              <Grid
                bsClass="container"
                className="divNoPadding"
                componentClass="div"
                fluid={false}
              >
                <Row
                  bsClass="row"
                  componentClass="div"
                >
                  <Col
                    bsClass="col"
                    componentClass="div"
                    lg={6}
                    md={6}
                  >
                    <Uncontrolled(Panel)>
                      <PanelHeading
                        bsClass="panel"
                        componentClass="div"
                      >
                        <PanelTitle
                          bsClass="panel"
                          componentClass="div"
                        >
                          If successful
                        </PanelTitle>
                      </PanelHeading>
                      <PanelBody
                        bsClass="panel"
                        className="centerStyle"
                        collapsible={false}
                      >
                        <img
                          alt="Login successful dialog"
                          src="successful-login.png"
                          style={
                            Object {
                              "align": "center",
                              "width": "300px",
                            }
                          }
                        />
                      </PanelBody>
                    </Uncontrolled(Panel)>
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass="div"
                    lg={6}
                    md={6}
                  >
                    <Uncontrolled(Panel)>
                      <PanelHeading
                        bsClass="panel"
                        componentClass="div"
                      >
                        <PanelTitle
                          bsClass="panel"
                          componentClass="div"
                        >
                          If unsuccessful
                        </PanelTitle>
                      </PanelHeading>
                      <PanelBody
                        bsClass="panel"
                        className="centerStyle"
                        collapsible={false}
                      >
                        <img
                          alt="Login unsuccessful dialog"
                          src="unsuccessful-dialog.png"
                          style={
                            Object {
                              "align": "center",
                              "width": "320px",
                            }
                          }
                        />
                      </PanelBody>
                    </Uncontrolled(Panel)>
                  </Col>
                </Row>
              </Grid>
            </div>
          </div>
          <BackButton
            link="/help/account"
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
